client/src/components/ModelViewerComponent/ModelViewerComponent.test.js

```diff
@@ -0,0 +1,307 @@
+  });
+
+  await findByText(container, "Floor");
+  const button = await findByLabelText(container, "modelViewerCommandBarComponent.farItems.uploadModel.text");
+  expect(uploadModel).toHaveBeenCalledTimes(0);
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const str = JSON.stringify(uploadValue);
+  const blob = new Blob([ str ]);
+  const file = new File([ blob ], "values.json", {
+    type: "application/json"
+  });
+  File.prototype.text = jest.fn().mockResolvedValueOnce(str);
+  const input = container.querySelectorAll(".mv-fileInput");
+  userEvent.upload(input[0], file);
+  await new Promise(resolve => setTimeout(resolve, 100));
+  expect(uploadModel).toHaveBeenCalledTimes(1);
+});
+
+
+
+test("upload model image", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  settingsService.setModelImage.mockResolvedValue(mockSuccesResponse);
+  act(() => {render(<ModelViewerComponent showItemMenu="true" />, container);});
+
+  await findByText(container, "Floor");
+  const button = container.querySelector(".ms-CommandBar-overflowButton");
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const options = await screen.findByLabelText("modelViewerItemCommandBarComponent.farItems.uploadModel");
+  act(() => {
+    options.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const str = JSON.stringify(uploadValue);
+  const blob = new Blob([str]);
+  const file = new File([blob], "Floor.json", {
+    type: "application/JSON",
+  });
+  File.prototype.text = jest.fn().mockResolvedValueOnce(str);
+  const input = container.querySelectorAll(".mv-fileInput");
+  userEvent.upload(input[2], file);
+  expect(retrieveModels).toHaveBeenCalledTimes(2);
+});
+
+test("delete model image", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  settingsService.getModelImage.mockResolvedValue(mockGetModelImage);
+  settingsService.deleteModelImage.mockResolvedValue(mockSuccesResponse);
+  act(() => {render(<ModelViewerComponent showItemMenu="true" />, container);});
+
+  await findByText(container, "Floor");
+  expect(getModelImage).toHaveBeenCalledTimes(1);
+  const button = container.querySelector(".ms-CommandBar-overflowButton");
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const options = await screen.findByLabelText("modelViewerItemCommandBarComponent.farItems.uploadModel");
+  act(() => {
+    options.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const deleteButton = await screen.getByTestId("deleteModelImage");
+  act(() => {
+    deleteButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  expect(retrieveModels).toHaveBeenCalledTimes(1);
```

Comment:
Would be great to validate that a delete function in the API service has been called here?